src/rules/exports-subpaths-style.ts

```diff
@@ -0,0 +1,149 @@
+import type { AST as JsonAST } from "jsonc-eslint-parser";
+
+import { createRule } from "../createRule.ts";
+
+type ExportsValue = JsonAST.JSONObjectExpression | JsonAST.JSONStringLiteral;
+
+function hasMultipleSubpaths(node: JsonAST.JSONObjectExpression): boolean {
+	const subpaths = node.properties.filter(
+		(property) =>
+			property.key.type === "JSONLiteral" &&
+			typeof property.key.value === "string" &&
+			property.key.value.startsWith("."),
+	);
+
+	return subpaths.length > 1;
+}
+
+function isImplicitFormat(node: ExportsValue): boolean {
+	if (node.type === "JSONLiteral") {
+		return true;
+	}
+
+	const keys = node.properties
+		.filter((property) => property.key.type === "JSONLiteral" && typeof property.key.value === "string")
+		.map((property) => (property.key as JsonAST.JSONStringLiteral).value);
```

Comment:
There's a type predicate function for `JSONStringLiteral` [here](https://github.com/JoshuaKGoldberg/eslint-plugin-package-json/blob/main/src/utils/predicates.ts#L3-L7) that should help narrow the type and avoid the casting.